Adding support for HttpPipeline

salaboy opened the pull request:
# Description

Adding support for HttpPipeline to DaprContainer

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1584

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

salaboy committed:
adding support for HttpPipeline

Signed-off-by: salaboy <Salaboy@gmail.com>

salaboy committed:
using default constructor

Signed-off-by: salaboy <Salaboy@gmail.com>

cicoyle reviewed (changes requested):
CI looks to be all red, mind taking a look and fixing up CI to be green again?

salaboy committed:
fixing codestyle

Signed-off-by: salaboy <Salaboy@gmail.com>

mcruzdev committed:
Add statestore example with Outbox pattern (#1582)

* Add statestore example with Outbox pattern

Signed-off-by: Matheus Cruz <matheuscruz.dev@gmail.com>

* Clean events after each test

Signed-off-by: Matheus Cruz <matheuscruz.dev@gmail.com>

* Add license header

Signed-off-by: Matheus Cruz <matheuscruz.dev@gmail.com>

* Apply pull request suggestions

Signed-off-by: Matheus Cruz <matheuscruz.dev@gmail.com>

---------

Signed-off-by: Matheus Cruz <matheuscruz.dev@gmail.com>
Co-authored-by: salaboy <Salaboy@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

salaboy committed:
adding new method signature plus test (#1570)

* adding new method signature plus test

Signed-off-by: salaboy <Salaboy@gmail.com>

* re adding imports

Signed-off-by: salaboy <Salaboy@gmail.com>

* fixing style

Signed-off-by: salaboy <Salaboy@gmail.com>

* checking empty metadata

Signed-off-by: salaboy <Salaboy@gmail.com>

* copy meta for safety and check if key is present

Signed-off-by: salaboy <Salaboy@gmail.com>

* Centralize Maven dependency version management (#1564)

Signed-off-by: salaboy <Salaboy@gmail.com>

* Fix dependencies multi app build and add proper test deps (#1572)

* Force Jackson version to override the SB Jackson version

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Move all the Jackson deps to parent POM.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Ensure app JAR build order

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Remove explicit Jackson from sdk-tests module.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Make sure <scope>test</scope> is used for test dependencies.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Remove extra Jackson modules.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

---------

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

* reverting pom

Signed-off-by: salaboy <Salaboy@gmail.com>

* fix codestyle

Signed-off-by: salaboy <Salaboy@gmail.com>

* using metaCopy

Signed-off-by: salaboy <Salaboy@gmail.com>

---------

Signed-off-by: salaboy <Salaboy@gmail.com>
Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>
Co-authored-by: artur-ciocanu <artur.ciocanu@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

cicoyle reviewed (dismissed):
Im good with the changes here, but can you add the tests to make codecov pass?

dependabot[bot] committed:
Bump actions/upload-artifact from 4 to 5 (#1587)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 4 to 5.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

wlfgang committed:
Add gRPC support to Dapr testcontainer (#1586)

* Add gRPC support to Dapr testcontainer

Signed-off-by: wlfgang <wlfgang@westridgesystems.com>

* Avoid using null to indicate default value

Signed-off-by: wlfgang <wlfgang@westridgesystems.com>

---------

Signed-off-by: wlfgang <wlfgang@westridgesystems.com>
Co-authored-by: artur-ciocanu <artur.ciocanu@gmail.com>
Co-authored-by: wlfgang <wlfgang@westridgesystems.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

artur-ciocanu committed:
Use dependencies BOM and remove duplicates. (#1588)

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

cicoyle committed:
Examples + Docs for App API Token authentication for gRPC and HTTP (#1589)

* example

Signed-off-by: Cassandra Coyle <cassie@diagrid.io>

* docs for example

Signed-off-by: Cassandra Coyle <cassie@diagrid.io>

---------

Signed-off-by: Cassandra Coyle <cassie@diagrid.io>
Signed-off-by: salaboy <Salaboy@gmail.com>

artur-ciocanu committed:
Another set of Maven version, properties and plugin improvements (#1596)

Signed-off-by: salaboy <Salaboy@gmail.com>

artur-ciocanu committed:
Adding a Flux based subscribeToEvents method (#1598)

* Adding a Flux based subscribeToEvents method

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Simplify GRPC stream handling

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Simplify Javadoc

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Fix unit tests and simplify implementation

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Adding event subscriber stream observer to simplify subscription logic

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Use start() method to start stream subscription

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Add unit test for event suscriber observer

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Improve the tests a little bit

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Remove the unnecessary method

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Improve error handling and use CloudEvent wrapper

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Fix unit tests asserts

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Adjust Java examples for Subscriber

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

---------

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

marcduiker committed:
Remove SDK docs due to migration to main Docs repo (#1593)

* Remove SDK docs due to migration to main Docs repo

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

* Remove sed lines related to sdk docs

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

---------

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>
Co-authored-by: salaboy <Salaboy@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

salaboy committed:
Bringing Durable Task Java as a Maven module inside the Java SDK (#1575)

* fixing checkstyle and javadocs

Signed-off-by: salaboy <Salaboy@gmail.com>

* Replace openjdk:17-jdk-slim to eclipse-temurin:17-jdk-jammy (#1574)

Signed-off-by: Matheus Cruz <matheuscruz.dev@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

* Align Java API with other languages (#1560)

* Align Java API with other languages

Signed-off-by: Matheus Cruz <matheuscruz.dev@gmail.com>

* Update documentation

Signed-off-by: Matheus Cruz <matheuscruz.dev@gmail.com>

* Change return type of waitForWorkflowStart method

Signed-off-by: artur-ciocanu <artur.ciocanu@gmail.com>

---------

Signed-off-by: Matheus Cruz <matheuscruz.dev@gmail.com>
Signed-off-by: artur-ciocanu <artur.ciocanu@gmail.com>
Co-authored-by: artur-ciocanu <artur.ciocanu@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

* use built in durable task

Signed-off-by: salaboy <Salaboy@gmail.com>

* exclude jacoco rules for examples and durabletask-client

Signed-off-by: salaboy <Salaboy@gmail.com>

* increasing timeout for IT

Signed-off-by: salaboy <Salaboy@gmail.com>

* removing dt build from matrix

Signed-off-by: salaboy <Salaboy@gmail.com>

* adding java to dt build

Signed-off-by: salaboy <Salaboy@gmail.com>

* Fix dependencies multi app build and add proper test deps (#1572)

* Force Jackson version to override the SB Jackson version

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Move all the Jackson deps to parent POM.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Ensure app JAR build order

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Remove explicit Jackson from sdk-tests module.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Make sure <scope>test</scope> is used for test dependencies.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Remove extra Jackson modules.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

---------

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

* docs: add architecture diagram to README (#1549)

* Preview New README

* Preview New README 2

* Preview New README 3

* docs: add architecture diagram showing Java SDK interaction with Dapr runtime (close #<915>)

* docs: add architecture diagram showing Java SDK interaction with Dapr runtime CORRECTION (close #<915>)

* docs: add architecture diagram showing Java SDK interaction with Dapr runtime (close #<915>)

* docs: add architecture diagram showing Java SDK interaction with Dapr runtime (close #<915>)

---------

Co-authored-by: Siri Varma Vegiraju <siri.varma@outlook.com>
Co-authored-by: artur-ciocanu <artur.ciocanu@gmail.com>
Co-authored-by: Cassie Coyle <cassie.i.coyle@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

* Add statestore example with Outbox pattern (#1582)

* Add statestore example with Outbox pattern

Signed-off-by: Matheus Cruz <matheuscruz.dev@gmail.com>

* Clean events after each test

Signed-off-by: Matheus Cruz <matheuscruz.dev@gmail.com>

* Add license header

Signed-off-by: Matheus Cruz <matheuscruz.dev@gmail.com>

* Apply pull request suggestions

Signed-off-by: Matheus Cruz <matheuscruz.dev@gmail.com>

---------

Signed-off-by: Matheus Cruz <matheuscruz.dev@gmail.com>
Co-authored-by: salaboy <Salaboy@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

* adding new method signature plus test (#1570)

* adding new method signature plus test

Signed-off-by: salaboy <Salaboy@gmail.com>

* re adding imports

Signed-off-by: salaboy <Salaboy@gmail.com>

* fixing style

Signed-off-by: salaboy <Salaboy@gmail.com>

* checking empty metadata

Signed-off-by: salaboy <Salaboy@gmail.com>

* copy meta for safety and check if key is present

Signed-off-by: salaboy <Salaboy@gmail.com>

* Centralize Maven dependency version management (#1564)

Signed-off-by: salaboy <Salaboy@gmail.com>

* Fix dependencies multi app build and add proper test deps (#1572)

* Force Jackson version to override the SB Jackson version

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Move all the Jackson deps to parent POM.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Ensure app JAR build order

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Remove explicit Jackson from sdk-tests module.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Make sure <scope>test</scope> is used for test dependencies.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Remove extra Jackson modules.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

---------

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

* reverting pom

Signed-off-by: salaboy <Salaboy@gmail.com>

* fix codestyle

Signed-off-by: salaboy <Salaboy@gmail.com>

* using metaCopy

Signed-off-by: salaboy <Salaboy@gmail.com>

---------

Signed-off-by: salaboy <Salaboy@gmail.com>
Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>
Co-authored-by: artur-ciocanu <artur.ciocanu@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

* Bump actions/upload-artifact from 4 to 5 (#1587)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 4 to 5.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

* Add gRPC support to Dapr testcontainer (#1586)

* Add gRPC support to Dapr testcontainer

Signed-off-by: wlfgang <wlfgang@westridgesystems.com>

* Avoid using null to indicate default value

Signed-off-by: wlfgang <wlfgang@westridgesystems.com>

---------

Signed-off-by: wlfgang <wlfgang@westridgesystems.com>
Co-authored-by: artur-ciocanu <artur.ciocanu@gmail.com>
Co-authored-by: wlfgang <wlfgang@westridgesystems.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

* Use dependencies BOM and remove duplicates. (#1588)

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

* Examples + Docs for App API Token authentication for gRPC and HTTP (#1589)

* example

Signed-off-by: Cassandra Coyle <cassie@diagrid.io>

* docs for example

Signed-off-by: Cassandra Coyle <cassie@diagrid.io>

---------

Signed-off-by: Cassandra Coyle <cassie@diagrid.io>
Signed-off-by: salaboy <Salaboy@gmail.com>

* Another set of Maven version, properties and plugin improvements (#1596)

Signed-off-by: salaboy <Salaboy@gmail.com>

* Adding a Flux based subscribeToEvents method (#1598)

* Adding a Flux based subscribeToEvents method

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Simplify GRPC stream handling

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Simplify Javadoc

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Fix unit tests and simplify implementation

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Adding event subscriber stream observer to simplify subscription logic

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Use start() method to start stream subscription

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Add unit test for event suscriber observer

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Improve the tests a little bit

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Remove the unnecessary method

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Improve error handling and use CloudEvent wrapper

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Fix unit tests asserts

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Adjust Java examples for Subscriber

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

---------

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

* Remove SDK docs due to migration to main Docs repo (#1593)

* Remove SDK docs due to migration to main Docs repo

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

* Remove sed lines related to sdk docs

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

---------

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>
Co-authored-by: salaboy <Salaboy@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

* cleaning up sdk version script

Signed-off-by: salaboy <Salaboy@gmail.com>

---------

Signed-off-by: salaboy <Salaboy@gmail.com>
Signed-off-by: Matheus Cruz <matheuscruz.dev@gmail.com>
Signed-off-by: artur-ciocanu <artur.ciocanu@gmail.com>
Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>
Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: wlfgang <wlfgang@westridgesystems.com>
Signed-off-by: Cassandra Coyle <cassie@diagrid.io>
Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>
Co-authored-by: Matheus Cruz <56329339+mcruzdev@users.noreply.github.com>
Co-authored-by: artur-ciocanu <artur.ciocanu@gmail.com>
Co-authored-by: Raymundo Zamora <raymundo.zamora@encora.com>
Co-authored-by: Siri Varma Vegiraju <siri.varma@outlook.com>
Co-authored-by: Cassie Coyle <cassie.i.coyle@gmail.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: wlfgang <14792753+wlfgang@users.noreply.github.com>
Co-authored-by: wlfgang <wlfgang@westridgesystems.com>
Co-authored-by: Marc Duiker <marcduiker@users.noreply.github.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

dependabot[bot] committed:
Bump codecov/codecov-action from 5.5.1 to 5.5.2 (#1607)

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from 5.5.1 to 5.5.2.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/v5.5.1...v5.5.2)

---
updated-dependencies:
- dependency-name: codecov/codecov-action
  dependency-version: 5.5.2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

artur-ciocanu committed:
Create Dapr WaitStrategy to improve ITs ergonomics (#1609)

* Create Dapr WaitStrategy to improve ITs ergonomics

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Improve unit tests naming and coverage

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Fix a potential NPE and remove extra "for..." methods for pubsub and topic.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Fix NPE properly for actor match

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Fix NPE another potential NPEs.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Rename to use wait strategy to use abstract prefix.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Add more tests to cover null checks

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Fix pubsub outbox IT.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Ignore pubsub outbox for now.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

* Disable pubsub outbox for now.

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>

---------

Signed-off-by: Artur Ciocanu <artur.ciocanu@gmail.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

dependabot[bot] committed:
Bump actions/upload-artifact from 4 to 6 (#1606)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 4 to 6.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v4...v6)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Cassie Coyle <cassie.i.coyle@gmail.com>
Co-authored-by: Dapr Bot <56698301+dapr-bot@users.noreply.github.com>
Signed-off-by: salaboy <Salaboy@gmail.com>

salaboy committed:
updating test for checking nulls

Signed-off-by: salaboy <Salaboy@gmail.com>

salaboy committed:
Merge branch 'master' into 1584-middleware

cicoyle reviewed (changes requested):
diff lgtm, but the history is showing a lot more commits than it should - can you clean up the history to just show your added commits from adding the logic in the PR?

salaboy committed:
Merge branch 'master' into 1584-middleware

siri-varma committed:
Merge branch 'master' into 1584-middleware
